components/net/http_loader.rs

```diff
@@ -568,6 +573,13 @@ pub fn http_fetch(
         // TODO(#21256) maybe set redirect_start if this resource initiates the redirect
         // TODO(#21254) also set startTime equal to either fetch_start or redirect_start
         //   (https://w3c.github.io/resource-timing/#dfn-starttime)
+        if request.redirect_count > 0 && request.headers.contains_key(header::LOCATION) {
```

Comment:
We want to check the response headers, not the request ones. However, I think if we add this `set_attribute` call to the `http_redirect_fetch` function, that should automatically cover the cases we care about.

Reply:
I do not think the header check is needed. If we unconditionally update fetchStart each time http_redirect_fetch is called, then the last updated value will be the correct one.